Source/engine/animationinfo.h

```diff
@@ -62,6 +62,18 @@ class AnimationInfo {
 	 */
 	bool IsPetrified;
 
+	uint8_t triggerFrame;
+
+	uint8_t lastFrame;
+
+	bool isLastFrame() const;

```

Comment:
An example of what I meant in the other comment, you can declare and define functions in the header itself. Using constexpr is mainly a helper for optimisation passes (adds restrictions to the function such that it can't branch and a few other things):
```suggestion
	constexpr bool isLastFrame() const
	{
		return (CurrentFrame == lastFrame);
	}
```

but it's not particularly well supported in old GCC versions, so sometimes you have to drop that keyword and just use:
```suggestion
	bool isLastFrame() const
	{
		return (CurrentFrame == lastFrame);
	}
```

You might also see the keyword `inline` used in other headers, this is already implicitly applied to class member functions defined inside the class definition so doesn't need to be explicitly written here.

Reply:
Thank you for short explanation. I will make them defined in header then :)